dlt/common/validation.py

```diff
@@ -50,9 +50,15 @@ def validate_dict(spec: Type[_TypedDict], doc: StrAny, path: str, filter_f: TFil
 
     def verify_prop(pk: str, pv: Any, t: Any) -> None:
         if is_optional_type(t):
+            # pass if value actually is none
```

Comment:
huh I have plenty of places where optional types are not present and all works! how it could work before? IMO something is wrong here

Reply:
I have changed this whole section to support union types properly, I would say it is correct now, I really don't know why it worked before...

Reply:
just wow